documentation/release-engineering.md

```diff
@@ -23,12 +23,10 @@ git commit -a -m "CHORE: Update dependencies"
 ```shell
 git checkout main
 git pull
-go fmt ./...
-bin/fmtjson $(find . -type f -name \*.json -print)
-for i in pkg/js/parse_tests/*.js ; do dnscontrol fmt -i $i -o $i ; done
-go generate ./...
-go mod tidy
+git checkout -b generate
+bin/generate-all.sh
 git status
+git commit -a -m "CHORE: generate-all.sh"
```

Comment:
I noticed that you’re using `git commit -a`, which automatically stages and commits all changes in the working directory. One potential risk here is that any open stashed changes might unintentionally be included in the commit if they are applied beforehand.

To prevent this, I suggest running `git reset --hard` before starting the generation process. This ensures that the working directory is in a clean state, reducing the chance of unintended changes being committed.

Let me know what you think!

```markdown
git fetch origin main
git reset --hard origin/main
git checkout -b generate
bin/generate-all.sh
git status
git commit -am "CHORE: generate-all.sh"
```